cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/Scheduler.java

```diff
@@ -80,16 +93,37 @@
    * @param program the program for which schedule needs to be resumed
    * @param programType the type of the program
    * @param scheduleName the name of the schedule
+   * @throws NotFoundException if the {@code scheduleName} could not be found, or if the application the {@code program}
+   *                           belongs to does not exist.
+   * @throws SchedulerException on unforeseen error.
    */
-  public void resumeSchedule(Id.Program program, SchedulableProgramType programType, String scheduleName);
+  public void resumeSchedule(Id.Program program, SchedulableProgramType programType, String scheduleName)
+    throws NotFoundException, SchedulerException;
+
+  /**
+   * Update the given schedule. The schedule with the same name than the given {@code schedule} will be replaced.
+   *
+   * @param program the program for which schedule needs to be updated
+   * @param programType the type of the program
+   * @param schedule the new schedule. The schedule with the same name will be replaced
+   * @throws NotFoundException if the {@code schedule} does not exist, or if the application the {@code program} 
```

Comment:
I thought the DefaultAppConfigurer validated that, but I checked and it doesn't, so I will make that extra validation in the deployment stage too.